### DISCUSSION TITLE
Allow plausible to reside in url folder rather than subdomain

### DISCUSSION THREAD

**Discussion by jbreu** [plausible/analytics#537] [Self-Hosted Support] [answered]
I want to host plausible under https://example.org/plausible and set the BASE_URL accordingly. However, it seems that such folders are not supported yet. All documentation is talking about "folder-less" domain or subdomain hosting.

I used https://plausible.io/docs/self-hosting to set everything up, including reverse proxy. I use reverse proxies for various other tools, so I am quite sure that is working all right.

Following screenshot shows that page assets are loaded folder-less, even though BASE_URL is set accordingly:

![grafik](https://user-images.githubusercontent.com/20864561/103364607-845f1e80-4abe-11eb-82b1-e6a4cefdee00.png)

I guess the base_url parameter needs to be systematically placed in every place where resources are called.

**Comment by Zulcom** [comment]
+1! Seems like Plausible ignores `BASE_URL`  config at all and gets only host from it. It's generally unusable under subfolder, system can't run under directory

**Comment by ukutaht** [comment, marked as answer]
Hi, thanks for reporting. This should be fixed in https://github.com/plausible/analytics/releases/tag/v1.4.0-rc.0. Can someone test and confirm?

**Reply by jbreu** [reply]
Saw it in the release note, great! Will try it asap

**Reply by jbreu** [reply]
Its a bit of progress for me, but not yet working:
On my instance with the new v1.4.0-rc.0 and proper reverse proxy config the login screen now works if I enter example.org/plausible/login
![grafik](https://user-images.githubusercontent.com/20864561/138749704-414c85b8-1299-40ac-a5b0-e1d0a4f87d24.png)
However, if I enter example.org/plausible/ or use my credentials on the login form, it always routes me to example.org/login (without plausible folder)
It seems BASE_URL is still not used throughout the whole application.

**Reply by ukutaht** [reply]
Ah I see. We have to change every single place we build URLs to also use the helpers provided by Phoenix.

There are hundreds of places where code needs to change. I'll consider doing it for this release but I won't promise anything at the moment, sorry.

**Reply by jbreu** [reply]
Yeah I already suspected that https://github.com/plausible/analytics/commit/85fcdbdbd6c1761350c66cb93ef15f2b32823997 alone won't do it. I am not really of help here, however I really appreciate your efforts! Thanks

**Reply by ukutaht** [reply]
We also have some issues on the client side with preserving the base path, see: https://github.com/plausible/analytics/issues/1038

Thanks for understanding. This will get fixed eventually but it's probably a bit too much to fit into this release.

**Comment by prairiewest** [comment]
I ran into the same thing: I want Plausible to live as a sub-folder of the main site, but it's really running behind a proxy to a private server behind firewall.

My setup:
Client -> example.com/stats/ -> Apache -> mod_proxy -> Plausible on host 192.168.1.2:8087

This does work on 1.4-rc.0 but requires some rewriting of HTML responses back to the client.  Many of the application links can be rewritten, but some are only assembled at runtime on the client side, so they cannot be rewritten and require additional Apache regex rewrite rules.

In case it helps @jbreu here are the rewrite rules working for me, attached.  As you can see, the top level "/api" is really the only outstanding problem.  Since we do not run anything at example.com/api this works for us, but for others who do have something already at /api they may have a conflict.  Also I have the BASE_URL set for Plausible in this case to "http://192.168.1.2:8087/"

[virtualhost.txt](https://github.com/plausible/analytics/files/7420566/virtualhost.txt)

**Reply by jbreu** [reply]
Thanks @prairiewest for your suggestions. I have set everything accordingly, but it didnt work for some reason (login screen is just fine, everything beyond shows 502 errors).

**Reply by prairiewest** [reply]
Did you enable apache module proxy_html? `a2enmod proxy_html` It's the one responsible for rewriting URLs in the HTML body.

**Reply by jbreu** [reply]
Hi @prairiewest , yes I did that, but it didnt help unfortunately.

**Comment by sebastian-raubach** [comment]
Is there any update on this? I'm trying to get Plausible CE running on something like https://example.com/plausible, but I immediately get redirected to https://example.com/register on launch.

**Reply by ruslandoga** [reply]
👋 @sebastian-raubach 

As of v2.1.1 it hasn't been implemented yet. The current solution might be to use a path-rewriting reverse proxy.

**Reply by sebastian-raubach** [reply]
Thanks for the response. Would you have any pointers as to how I can achieve that with nginx which I'm currently using as a reverse proxy already?

**Reply by ruslandoga** [reply]
Actually, ignore what I said about it being possible. It's not possible right now since all the links that Plausible generates would stay "unscoped" in HTML, like `href=/register` instead of `href=/plausible/register`. I'll ask the core team about providing a "scope" to the URL builder in Plausible to make this use case possible. Thank you for bringing this up!

**Comment by mqu** [comment]
+1 : I am also interested in running Plausible in sub-URL in the form : https://my.app/stats/
